JacksonDunstanNativeCollections/NativePerJobThreadIntPtrs.cs

```diff
@@ -0,0 +1,434 @@
+			{
+				RequireWriteAccess();
+				m_Buffer[IntsPerCacheLine * m_ThreadIndex + index]++;
+			}
+
+			/// <summary>
+			/// Decrement the stored value
+			/// </summary>
+			/// 
+			/// <returns>
+			/// This object
+			/// </returns>
+			[WriteAccessRequired]
+			public void Decrement(int index)
+			{
+				RequireWriteAccess();
+				m_Buffer[IntsPerCacheLine * m_ThreadIndex + index]--;
+			}
+
+			/// <summary>
+			/// Add to the stored value
+			/// </summary>
+			/// 
+			/// <param name="value">
+			/// Value to add. Use negative values for subtraction.
+			/// </param>
+			/// 
+			/// <returns>
+			/// This object
+			/// </returns>
+			[WriteAccessRequired]
+			public void Add(int index, int value)
+			{
+				RequireWriteAccess();
+				m_Buffer[IntsPerCacheLine * m_ThreadIndex + index] += value;
+			}
+
+			/// <summary>
+			/// Throw an exception if the object isn't writable
+			/// </summary>
+			[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+			[BurstDiscard]
+			private void RequireWriteAccess()
+			{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+				AtomicSafetyHandle.CheckWriteAndThrow(m_Safety);
+#endif
+			}
+		}
+
+		/// <summary>
+		/// The number of int ptrs.
+		/// </summary>
+		public int Length;
+
+		/// <summary>
+		/// Pointer to the value in native memory. Must be named exactly this
+		/// way to allow for [NativeContainerSupportsDeallocateOnJobCompletion]
+		/// </summary>
+		[NativeDisableUnsafePtrRestriction]
+		internal int* m_Buffer;
+
+		/// <summary>
+		/// Allocator used to create the backing memory
+		/// 
+		/// This field must be named this way to comply with
+		/// [NativeContainerSupportsDeallocateOnJobCompletion]
+		/// </summary>
+		internal Allocator m_AllocatorLabel;
+
+		// These fields are all required when safety checks are enabled
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+		/// <summary>
+		/// A handle to information about what operations can be safely
+		/// performed on the object at any given time.
+		/// </summary>
+		private AtomicSafetyHandle m_Safety;
+
+		/// <summary>
+		/// A handle that can be used to tell if the object has been disposed
+		/// yet or not, which allows for error-checking double disposal.
+		/// </summary>
+		[NativeSetClassTypeToNullOnSchedule]
+		private DisposeSentinel m_DisposeSentinel;
+#endif
+
+		/// <summary>
+		/// The number of integers that fit into a CPU cache line
+		/// </summary>
+		private const int IntsPerCacheLine = JobsUtility.CacheLineSize / sizeof(int);
+
+		/// <summary>
+		/// The number of integers that fit into a CPU cache line
+		/// </summary>
+		private readonly int NumCacheLines;
+
+		/// <summary>
+		/// Allocate memory and set the initial value
+		/// </summary>
+		/// 
+		/// <param name="allocator">
+		/// Allocator to allocate and deallocate with. Must be valid.
+		/// </param>
+		/// 
+		/// <param name="initialValue">
+		/// Initial value of the allocated memory
+		/// </param>
+		public NativePerJobThreadIntPtrs(int length, Allocator allocator)
```

Comment:
`length` is always `IntsPerCacheLine` since this type is just trying to make more use of a single cache line, not provide an arbitrary length array. So this shouldn't be a parameter here or in `Parallel`. Similarly, there's no need for `NumCacheLines` since it is always `1`.

Reply:
Explained below why this is not constant.